File: src/org/dgraph/graph/edge/Edge.java
package org.dgraph.graph.edge;

/** Main interface for all edges in DGraph library.
 *
 * @param <V> type for vertices
 *
 * @author Andrii Dzhyrma
 * @since 0.1 */
public interface Edge<V> {

	/** Retrieves the source of this edge.
	 * 
	 * @return source of this edge */
	public V getSource();

	/** Retrieves the target of this edge.
	 * 
	 * @return target of this edge */
	public V getTarget();
}

File: src/main/kotlin/org/dzhyrma/dgraph/edge/Edge.kt
package org.dzhyrma.dgraph.edge

/**
 * Main interface for all edges.
 *
 * @param <V> type for vertices
 */
interface Edge<V> {

    /**
     * Returns the source of this edge.
     */
    val source: V

    /**
     * Returns the target of this edge.
     */
    val target: V
}

data class SimpleEdge<V>(
    override val source: V,
    override val target: V,
) : Edge<V>
